gcp-observability/src/test/java/io/grpc/gcp/observability/GcpObservabilityTest.java

```diff
@@ -18,48 +18,186 @@
 import io.grpc.gcp.observability.interceptors.InternalLoggingChannelInterceptor;
 import io.grpc.gcp.observability.interceptors.InternalLoggingServerInterceptor;
 import io.grpc.gcp.observability.logging.Sink;
+import io.opencensus.trace.samplers.Samplers;
+import java.io.IOException;
+import java.util.regex.Pattern;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
 @RunWith(JUnit4.class)
 public class GcpObservabilityTest {
-  
+
+  private final StaticTestingClassLoader classLoader =
+      new StaticTestingClassLoader(
+          getClass().getClassLoader(),
+          Pattern.compile(
+              "io\\.grpc\\.InternalGlobalInterceptors|io\\.grpc\\.GlobalInterceptors|"
+                  + "io\\.grpc\\.gcp\\.observability\\.[^.]+|"
+                  + "io\\.grpc\\.gcp\\.observability\\.interceptors\\.[^.]+|"
+                  + "io\\.grpc\\.gcp\\.observability\\.GcpObservabilityTest\\$.*"));
+
+  @Test
+  public void initFinish() throws Exception {
+    Class<?> runnable =
+        classLoader.loadClass(StaticTestingClassInitFinish.class.getName());
+    ((Runnable) runnable.getDeclaredConstructor().newInstance()).run();
+  }
+
+  @Test
+  public void enableObservability() throws Exception {
+    Class<?> runnable =
+        classLoader.loadClass(StaticTestingClassEnableObservability.class.getName());
+    ((Runnable) runnable.getDeclaredConstructor().newInstance()).run();
+  }
+
   @Test
-  public void initFinish() {
-    ManagedChannelProvider prevChannelProvider = ManagedChannelProvider.provider();
-    ServerProvider prevServerProvider = ServerProvider.provider();
-    Sink sink = mock(Sink.class);
-    InternalLoggingChannelInterceptor.Factory channelInterceptorFactory = mock(
-        InternalLoggingChannelInterceptor.Factory.class);
-    InternalLoggingServerInterceptor.Factory serverInterceptorFactory = mock(
-        InternalLoggingServerInterceptor.Factory.class);
-    GcpObservability observability1;
-    try (GcpObservability observability = GcpObservability.grpcInit(sink, channelInterceptorFactory,
-        serverInterceptorFactory)) {
-      assertThat(ManagedChannelProvider.provider()).isInstanceOf(LoggingChannelProvider.class);
-      assertThat(ServerProvider.provider()).isInstanceOf(ServerProvider.class);
-      observability1 = GcpObservability.grpcInit(sink, channelInterceptorFactory,
-              serverInterceptorFactory);
-      assertThat(observability1).isSameInstanceAs(observability);
+  public void disableObservability() throws Exception {
+    Class<?> runnable =
+        classLoader.loadClass(StaticTestingClassDisableObservability.class.getName());
+    ((Runnable) runnable.getDeclaredConstructor().newInstance()).run();
+  }
+
+  // UsedReflectively
+  public static final class StaticTestingClassInitFinish implements Runnable {
+
+    @Override
+    public void run() {
+      ManagedChannelProvider prevChannelProvider = ManagedChannelProvider.provider();
```

Comment:
With your next PR the *Provider code is going to go away so this part of the test will also go away, right? May be add a TODO as a reminder for yourself for the next PR?

Reply:
Added TODO.

I think we would still need the test to check for an exception when `close()` is called twice on GcpObservability instance. Lmk if I am missing something.

Reply:
You are correct about "we would still need the test to check for an exception when close() is called twice"